multicluster/apis/multicluster/v1alpha1/clusterset_types.go

```diff
@@ -27,14 +27,23 @@ type MemberCluster struct {
 	ClusterID string `json:"clusterID,omitempty"`
 	// API server of the destination cluster.
 	Server string `json:"server,omitempty"`
-	// Secret name to access API server of the member from the leader cluster.
+	// Secret name in the member cluster which will be used to access API
+	// server of the leader cluster.
 	Secret string `json:"secret,omitempty"`
-	// ServiceAccount used by the member cluster to access into leader cluster.
+	// ServiceAccount indicates which account is bound by the member cluster's access token
```

Comment:
ServiceAccount in the leader cluster, from which the member cluster's token is generated.

Reply:
After reading through the types and their usage, I feel it is very confusing to share one struct for both leader cluster and member cluster. And I also found we actually do not use the `ClusterSetSpec.Members` field?

In this case, what you think we rename `MemberCluster` to `LeaderClusterInfo`, and deprecate `ClusterSetSpec.Members`.

Reply:
@jianjuns I feel we may rename `MemberCluster` to `Cluster` considering `ClusterID` field is actually used in both in leader and member cluster, below is the new structure, what's your thoughts?
```
// Cluster specifies information of a cluster, which can be a
// regular member cluster or a leader cluster of the ClusterSet.
type Cluster struct {
	// LeaderAccessInfo provides the access info to the leader cluster.
	// It's only required by a member cluster.
	LeaderAccessInfo
	// Identify a cluster in the ClusterSet.
	ClusterID string `json:"clusterID,omitempty"`
	// ServiceAccount in the leader cluster, from which the member cluster's token
	// is generated. This is an optional field which helps admin to check
	// which ServiceAccount is used by a member cluster to access the leader cluster.
	ServiceAccount string `json:"serviceAccount,omitempty"`
}

type LeaderAccessInfo struct {
	// API server endpoint of the leader cluster. Used only for a member cluster.
	// eg: "https://172.18.0.1:6443", "https://example.com:6443".
	Server string `json:"server,omitempty"`
	// Name of the Secret resource in the member cluster, which stores
	// the token to access the leader cluster's API server.
	Secret string `json:"secret,omitempty"`
}
```

Reply:
Another question about `deprecate ClusterSetSpec.Members`, I checked the codes, it's actually only used by memberclusterannounce_controller to remove member cluster if it's existing in the spec when memberclusterannounce is deleted, it's indeed not really used in ClusterSet setup.  Do you mean to remove this field to deprecate it?

Reply:
I didn't mean it's used for creating a ClusterSet, it's being used here when MemberClusterAnnounce is deleted https://github.com/antrea-io/antrea/blob/main/multicluster/controllers/multicluster/leader/memberclusterannounce_controller.go#L260-L287, I feel it's safe to delete the field considering it's not required for ClusterSet. Just like to double check with you if it's OK to remove this field directly to deprecate it.

Reply:
If it is not used, let us deprecate it. But we should not directly remove it. Just mark it as "deprecated" in comments, and remove it in the new API version.

Reply:
> I feel we may rename MemberCluster to Cluster considering ClusterID field is actually used in both in leader and member cluster

If `ClusterSetSpec.Members` is deprecated, when you will need the struct and ClusterID for a member cluster? The struct should be just for a leader cluster right?

Reply:
Done.
Yes, it's no longer used by a member cluster, so I keep the old struct and rename it as LeaderClusterInfo.